src/repositories/application-repository.js

```diff
@@ -6,11 +6,54 @@
 import { APPLICATION_COLLECTION } from '../constants/index.js'
 
 export const getApplication = async (db, reference) => {
-  const application = await db
+  return db
     .collection(APPLICATION_COLLECTION)
-    .findOne({ reference }, { projection: { _id: 0 } })
-
-  return application
+    .aggregate([
+      {
+        $match: {
+          reference: reference.toUpperCase()
+        }
+      },
+      {
+        $project: {
+          _id: 0,
+          reference: 1,
+          createdAt: 1,
+          updatedAt: 1,
+          createdBy: 1,
+          updatedBy: 1,
+          data: 1,
+          organisation: 1,
+          status: 1,
+          flags: {
+            $map: {
+              input: {
+                $filter: {
+                  input: '$flags',
+                  as: 'flag',
+                  cond: { $eq: ['$$flag.deletedBy', null] }
```

Comment:
The intention in migration is to set a boolean field, explicitly called **deleted**, which could be used for conditions like this. It might not be being set yet by the application layer, but should be when we get there